### ISSUE TITLE
Resnet50 for Imagenet

### ISSUE THREAD

**Issue by atrah22** [youzhonghui/gate-decorator-pruning#20]
Hello,

I am  getting an error on mismatched Batch normalization layer when pruning resnet50 for imagenet dataset.

Could you please provide Resnet50 model definition and how it is used for imagenet dataset pruning?

**Comment by youzhonghui** [comment]
Probably because some filters that did not work during the observe process were cut out by mistake. The easiest solution is to increase OBSERVE_TIMES in prune/universal.py line 13.

The definition of resnet50 is in models/imagenet/resnet50.py.

Our code to prune resnet50 can be download from here ->  [resnet50.zip](https://github.com/youzhonghui/gate-decorator-pruning/files/4316032/resnet50.zip). But due to the change of code base, the code may require some modifications to run. We also provide the output of the GBN-50 experiment.

**Comment by atrah22** [comment]
Thankyou

**Comment by atrah22** [comment]
Hello,
In your attached code, the pretrained resnet50 model is not loaded. And if it is loaded (I modified) then there is missing keys.
How did you load the pretrained resnet50 model. Didn't see any part where the pretrained model is loaded.
